sql/executor.go

```diff
@@ -41,8 +41,9 @@ var errTransactionInProgress = errors.New("there is already a transaction in pro
 
 // An Executor executes SQL statements.
 type Executor struct {
-	db     client.DB
-	nodeID uint32
+	db      client.DB
+	nodeID  uint32
+	reCache *parser.RegexpCache
```

Comment:
could you make this a value, rather than a pointer? you'll need to change RegexpCache.mu to `sync.Locker`.